controllers/openstackdataplane_controller.go

```diff
@@ -62,5 +218,97 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 func (r *OpenStackDataPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&dataplanev1beta1.OpenStackDataPlane{}).
+		Owns(&dataplanev1beta1.OpenStackDataPlaneNode{}).
+		Owns(&dataplanev1beta1.OpenStackDataPlaneRole{}).
 		Complete(r)
 }
+
+// CreateDataPlaneResources -
+func CreateDataPlaneResources(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlane, helper *helper.Helper) (ctrl.Result, error) {
+	// roles and nodes may depend on each other, so we run both before checking for errors
+	roleErr := CreateDataPlaneRole(ctx, instance, helper)
+	nodeErr := CreateDataPlaneNode(ctx, instance, helper)
+	if roleErr != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityError,
+			dataplanev1beta1.DataPlaneRoleErrorMessage,
+			roleErr.Error()))
+		return ctrl.Result{}, roleErr
+	}
+	if nodeErr != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityError,
+			dataplanev1beta1.DataPlaneNodeErrorMessage,
+			nodeErr.Error()))
+		return ctrl.Result{}, nodeErr
+	}
+
+	return ctrl.Result{}, nil
+
+}
+
+// CreateDataPlaneNode -
+func CreateDataPlaneNode(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlane, helper *helper.Helper) error {
+	client := helper.GetClient()
+	logger := helper.GetLogger()
+	node := &dataplanev1beta1.OpenStackDataPlaneNode{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: instance.Namespace,
+		},
+	}
```

Comment:
You need to move this node assignment into the loop below. What is causing the traceback is that it creates the first node, which sets a resourceVersion, then when it goes to create the 2nd node, it reuses this same var where resourceVersion is set.